includes/class-send-list.php

```diff
@@ -0,0 +1,213 @@
+					'type'     => 'string',
+					'required' => true,
+					'enum'     => [
+						'list',
+						'sublist',
+					],
+				],
+				// The type of entity this list or sublist is associated with in the ESP. Controls which ESP API endpoints to use to fetch/update.
+				'entity_type' => [
+					'name'     => 'entity_type',
+					'type'     => 'string',
+					'required' => true,
+				],
+				// The ID of the list or sublist as identified in the ESP.
+				'id'          => [
+					'name'     => 'id',
+					'type'     => 'string',
+					'required' => true,
+				],
+				'value'       => [
+					'name'     => 'value',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// The name of the list or sublist as identified in the ESP.
+				'name'        => [
+					'name'     => 'name',
+					'type'     => 'string',
+					'required' => true,
+				],
+				'label'       => [
+					'name'     => 'label',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If the list is also a Subscription List, it could have a locally edited name.
+				'local_name'  => [
+					'name'     => 'local_name',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If available, the number of contacts associated with this list or sublist.
+				'count'       => [
+					'name'     => 'count',
+					'type'     => 'integer',
+					'required' => false,
+				],
+				// If this Send_List is a sublist, this property must indicate the ID of the parent list.
+				'parent_id'   => [
+					'name'     => 'parent_id',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If it can be calculated, the URL to view this list or sublist in the ESP's dashboard.
+				'edit_link'   => [
+					'name'     => 'edit_link',
+					'type'     => 'string',
+					'required' => false,
+				],
+			],
+		];
+	}
+
+	/**
+	 * Get the Send_List configuration.
+	 *
+	 * @return array
+	 */
+	public function get_config() {
+		$schema = self::get_config_schema();
+		$config  = [];
+		foreach ( $schema['properties'] as $key => $property ) {
+			$config[ $key ] = $this->get( $key ) ?? null;
+		}
+
+		return array_filter( $config );
+	}
+
+	/**
+	 * Get a specific property's value.
+	 *
+	 * @param string $key The property to get.
+	 *
+	 * @return mixed The property value or null if not set/not a supported property.
+	 */
+	public function get( $key ) {
+		return $this->{ $key } ?? null;
+	}
+
+	/**
+	 * Set a property's value.
+	 *
+	 * @param string $key The property to get.
+	 * @param mixed  $value The value to set.
+	 *
+	 * @return mixed The property value or null if not set/not a supported property.
+	 */
+	public function set( $key, $value ) {
+		$schema = $this->get_config_schema();
+		if ( ! isset( $schema['properties'][ $key ] ) ) {
+			return null;
+		}
+		$this->{ $key } = $value;
+		return $this->get( $key );
+	}
+
+	/**
+	 * Set the label and value properties for autocomplete inputs.
+	 */
+	public function set_label_and_value() {
```

Comment:
Given this is always dynamic, based on the Send_List config, is there a specific reason we're setting and storing? A `get_label_and_value()` would work well and not require every config change to call this method for an update.

Reply:
I stored them as properties mainly because these are for the `Autocomplete` component in the editor JS, and having them automatically generated and set by the constructor means that I never have to make sure they're set before using them there. But you're right that if we ever update any of the properties that are used to generate these, we would need to regenerate the dynamic values. Let me think about this a bit more.

Reply:
7f4906e introduces `get_label()` and `get_value()` methods which will either retrieve the stored values (if passed in the constructor config) or dynamically generate the values. 7f4906e also takes inspiration from @leogermani's `Subscription_List` object with a `to_array()` method which will return the send list's data as an associative array. This gives us a chance to check whether we need to generate the dynamic values and append them to the response, and it also will let us tailor the data we pass back as REST responses (to be handled in #1631).

Reply:
If this isn't needed then we can remove `label` and `value` from the schema and remove all the conditionals. Otherwise we can keep as-is to retain the future option for manual overrides.